syntax = "proto3";

package TW.NEO.Proto;
option java_package = "wallet.core.jni.proto";

message TransactionInput {
    bytes prev_hash = 1;
    fixed32 prev_index = 2;
}

message TransactionOutput {
    string asset_id = 1;
    sint64 value = 2;
    string address = 3;
}

// Input data necessary to create a signed transaction.
message SigningInput {
    repeated TransactionInput inputs = 1;
    repeated TransactionOutput outputs = 2;
    bytes private_key = 3;
}

// Transaction signing output.
message SigningOutput {
    // Signed and encoded transaction bytes.
    bytes encoded = 1;
}
